src/main/java/org/jahia/modules/sitemap/utils/Utils.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * ==========================================================================================
+ * =                            JAHIA'S ENTERPRISE DISTRIBUTION                             =
+ * ==========================================================================================
+ *
+ *                                  http://www.jahia.com
+ *
+ * JAHIA'S ENTERPRISE DISTRIBUTIONS LICENSING - IMPORTANT INFORMATION
+ * ==========================================================================================
+ *
+ *     Copyright (C) 2002-2021 Jahia Solutions Group. All rights reserved.
+ *
+ *     This file is part of a Jahia's Enterprise Distribution.
+ *
+ *     Jahia's Enterprise Distributions must be used in accordance with the terms
+ *     contained in the Jahia Solutions Group Terms &amp; Conditions as well as
+ *     the Jahia Sustainable Enterprise License (JSEL).
+ *
+ *     For questions regarding licensing, support, production usage...
+ *     please contact our team at sales@jahia.com or go to http://www.jahia.com/license.
+ *
+ * ==========================================================================================
+ */
+package org.jahia.modules.sitemap.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URL;
+
+/**
+ * General utility class
+ */
+public class Utils {
+
+    private static final Logger logger = LoggerFactory.getLogger(Utils.class);
+
+    /**
+     * General utility method to get the server-name based on url
+     * In the-case it cannot be parsed to URL or it does not have protocol will return null
+     *
+     * @param urlString [String] url string
+     * @return  servername in format of protocal://hostname:port
+     */
+    public static String getServerName(String urlString) {
+        try {
```

Comment:
It might be better to use URI package to parse than roll our own implementation. See here: https://stackoverflow.com/a/9608008

Reply:
Based on the comments 
It seems like URI maybe handle some cases better (being said I believe the implementation is the same regardless if we use URL or URI)